File: DSAlgorithms/Leetcode/c++/2_pointers/167-2sum2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
       int left = 0;
       int right = numbers.size() - 1;
       std::vector<int> result;
       while(left < right) 
       {
           int current_sum = numbers[left] + numbers[right];
           if(current_sum > target)
                right--;
            else if(current_sum < target)
                left++;
            else
            {
                result.push_back(left + 1);
                result.push_back(right + 1);
                return result;
            }
       }
       return result;
    }
};

File: DSAlgorithms/Leetcode/ts/src/2_pointers/167-2sum2.ts
const twoSum2 = (numbers: number[], target: number): number[] =>{
    let left = 0;
    let right = numbers.length - 1;
    while(left < right)
    {
        let current_sum = numbers[left] + numbers[right];
        if(current_sum > target)
            right--;
        else if(current_sum < target)
            left++;
        else
            return [left + 1, right + 1]
    }
    return [];
};